File: source/scene/SceneSelectedData.java
package scene;

import camera.abs.Camera;
import entity.Entity;
import light.AmbientLight;
import light.DirectionalLight;

import java.io.Serializable;

public final class SceneSelectedData implements Serializable {
    public Entity entity = null;
    public Camera camera = null;
    public AmbientLight ambientLight = null;
    public DirectionalLight directionalLight = null;

    public SceneSelectedData() {}
}

File: source/scene/SceneSelectedData.kt
package scene

import camera.abs.Camera
import entity.Entity
import light.AmbientLight
import light.DirectionalLight
import java.io.Serializable

class SceneSelectedData : Serializable {
    var entity: Entity? = null
    var camera: Camera? = null
    var ambientLight: AmbientLight? = null
    var directionalLight: DirectionalLight? = null
}
